core/src/main/java/com/linecorp/armeria/client/AbstractHttpRequestHandler.java

```diff
@@ -179,8 +179,11 @@ final boolean tryInitialize() {
         if (timeoutMillis > 0) {
             // The timer would be executed if the first message has not been sent out within the timeout.
             timeoutFuture = ch.eventLoop().schedule(
-                    () -> failAndReset(WriteTimeoutException.get()),
-                    timeoutMillis, TimeUnit.MILLISECONDS);
+                    () -> {
+                        final WriteTimeoutException exception = WriteTimeoutException.get();
+                        failAndReset(exception);
+                        ctx.cancel(exception);
```

Comment:
Question) Is this necessary? Since `res.close` is called in the next line, I was assuming `AbstractRuleBuilder` will handle timed out cases anyways

https://github.com/line/armeria/blob/b604f3f31f4a571da7809e9d524d0b1576c840d4/core/src/main/java/com/linecorp/armeria/client/AbstractRuleBuilder.java#L181

I doubt we will remember to add this line to other future locations if another timeout behavior is added.
If it is necessary, can you add a test to guard against regressions?

Ditto for the line at `HttpClientDelegate`

Reply:
> I doubt we will remember to add this line to other future locations if another timeout behavior is added.

```java
// RequestContext.java
    default boolean isTimedOut() {
        return cause instanceof TimeoutException ||
         // 1. cause instanceof UnprocessedRequestException && cause.getCause() instanceof TimeoutException;
         // 2. (idea) we can add Exceptions.peel(cause) instanceof TimeoutException ..
```
I agree! @ikhoon nim, instead of adding `ctx.cancel(timeoutException)` on each location,
how about adding more detailed logic on `RequestContext.isTimedOut`?